core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/integration_test.go

```diff
@@ -534,7 +527,7 @@ func collectPayloads(ctx context.Context, t *testing.T, logProvider logprovider.
 	for ctx.Err() == nil && len(allPayloads) < n && rounds > 0 {
 		logs, err := logProvider.GetLatestPayloads(ctx)
 		require.NoError(t, err)
-		require.LessOrEqual(t, len(logs), logprovider.AllowedLogsPerUpkeep, "failed to get all logs")
+		require.LessOrEqual(t, len(logs), logprovider.AllowedLogsPerUpkeep)
```

Comment:
This is where `TestIntegration_LogEventProvider` fails. Is this a per-round assumption? Meaning if this loop went fast enough, then the test would pass?